grails-app/services/org/pih/warehouse/core/LocationService.groovy

```diff
@@ -96,6 +96,12 @@ class LocationService {
             locations = locations.findAll { location -> user.hasPrimaryRole(location) }
         }
 
+        if (params.type == "REQUEST") {
```

Comment:
Can we pass the activity code and deal with that directly? Remember that this API can be used by other clients, so randomly assigning a type that only applies for one API client seems like it might end up being more confusing than supporting a list of supported activities (activity codes). Also I'm implying that the activityCode parameter should support multiple values, if possible.